openprescribing/frontend/tests/test_models.py

```diff
@@ -249,3 +252,42 @@ def test_regional_team_with_no_pcts_excluded(self):
         # instances of each team and we want to ensure that this doesn't happen
         self.assertEqual(len(active_teams), len(all_teams) - 1)
         self.assertNotIn(empty_team, active_teams)
+
+
+class PPUSavingTests(TestCase):
+
+    def test_exists_for_bnf_code(self):
+        factory = DataFactory()
+        ccg = factory.create_ccg()
+
+        presentation = Presentation.objects.create(
+            bnf_code='0101010I0AAABAB',
+            name='Mag Ox_Cap 100mg',
+        )
+
+        PPUSaving.objects.create(
+            presentation=presentation,
+            pct=ccg,
+            date='2019-07-01',
+            lowest_decile=0.1,
+            quantity=10,
+            price_per_unit=20,
+            possible_savings=10,
+        )
+
+        GenericCodeMapping.objects.create(
+            from_code='0101010I0AAAEAE',
+            to_code='0101010I0AAABAB'
+        )
+
+        # Mag Ox_Cap 100mg
+        self.assertTrue(PPUSaving.objects.exists_for_bnf_code('0101010I0AAABAB'))
```

Comment:
Can we add a test for the code path where the mapping ends in a percent?

Reply:
Good spot.

Reply:
No need to do this now.